### PR TITLE
Do not attempt to guess the service in paasta <stop/restart/start> wh…

### LINKED ISSUE #588: paasta stop without `-i` tries to stop unrelated instance on different service
The only instance for the `error_pages` service is `main`. `ad_bulk_calculate_recommended_package_worker` is from some other unrelated service

```
$ paasta stop -s error_pages -c norcal-stagef
Error: error_pages.ad_bulk_calculate_recommended_package_worker doesn't look like it has been deployed to this cluster! (norcal-stagef)
```

### PR DESCRIPTION
**Pull request by ammaraskar**
…en getting the list of instances. Fixes #588

### CODE REVIEW

File: paasta_tools/cli/cmds/start_stop_restart.py [hunk 1]
```diff
@@ -150,8 +150,6 @@ def issue_state_change_for_service(service_config, force_bounce, desired_state):
 
 def paasta_start_or_stop(args, desired_state):
     """Requests a change of state to start or stop given branches of a service."""
-    instances = args.instances
```
Comment (ammaraskar) [inline paasta_tools/cli/cmds/start_stop_restart.py]: Removed because unused variables

File: paasta_tools/cli/cmds/start_stop_restart.py [hunk 2]
```diff
@@ -163,7 +161,7 @@ def paasta_start_or_stop(args, desired_state):
     if args.instances is not None:
         instances = args.instances.split(",")
     else:
-        instances = list_instances()
+        instances = list_instances(service=service)
```
Comment (solarkennedy) [inline paasta_tools/cli/cmds/start_stop_restart.py]: You need cluster or something here too, otherwise it will try to stop instances for other clusters?

**Comment by ammaraskar** [comment]
Updated, see the second commit.

**Comment by solarkennedy** [comment]
ship

**Comment by ammaraskar** [comment]
https://github.com/Yelp/paasta/commit/ee545aa8fd0b033f0e1900393de5f4c8a5682b6b

**Pull request closed without merging**